nss/integration-tests/nss_test.go

```diff
@@ -0,0 +1,176 @@
+package nss_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testutils"
+)
+
+var libPath string
+var rustCovEnv []string
+
+func TestIntegration(t *testing.T) {
+	t.Parallel()
+
+	buildRustNSSLib(t)
+
+	originOuts := map[string]string{}
+	for _, db := range []string{"passwd", "group", "shadow"} {
+		//#nosec:G204 - We control the cmd arguments in tests.
+		data, err := exec.Command("getent", db).CombinedOutput()
+		require.NoError(t, err, "Setup: can't run getent to get original output from system")
+		originOuts[db] = string(data)
+	}
+
+	tests := map[string]struct {
+		db      string
+		key     string
+		cacheDB string
+
+		noDaemon       bool
+		noCustomSocket bool
+
+		wantErr bool
+	}{
+		"Get all entries from passwd": {db: "passwd"},
+		"Get all entries from group":  {db: "group"},
+		"Get all entries from shadow": {db: "shadow"},
+
+		"Get entry from passwd by name": {db: "passwd", key: "user1"},
+		"Get entry from group by name":  {db: "group", key: "group1"},
+		"Get entry from shadow by name": {db: "shadow", key: "user1"},
+
+		"Get entry from passwd by id": {db: "passwd", key: "1111"},
+		"Get entry from group by id":  {db: "group", key: "11111"},
+
+		/* Error cases */
+		// Even though those are "error" cases, the getent command won't fail since the other databases on the machine will return some entries.
+		"Uses default socket if env is not set":                                          {db: "passwd", noCustomSocket: true},
+		"Returns empty when getting all entries from passwd and daemon is not available": {db: "passwd", noDaemon: true},
+		"Returns empty when getting all entries from group and daemon is not available":  {db: "group", noDaemon: true},
+		"Returns empty when getting all entries from shadow and daemon is not available": {db: "shadow", noDaemon: true},
+
+		"Returns empty when getting all entries from passwd and database is corrupted": {db: "passwd", cacheDB: "invalid_entry_in_userByID"},
```

Comment:
Getting all entries never returns an error, because it also returns the entries from other databases... I can add the second call, but the outcome will be the same (and the only way to check the test did the right thing is looking into the coverage)

Reply:
ack then.